### DISCUSSION TITLE
Gluetun Custom Wireguard with TorGuard Only Works Once?

### DISCUSSION THREAD

**Discussion by MrLeaw** [qdm12/gluetun#1745] [Q&A]
I created a wireguard config on the torguard website, and followed the wiki here using the custom wireguard section, and was excited to see that everything is working as expected. Then, I shut down my container to add a port to the firewall exceptions (FIREWALL_VPN_INPUT_PORTS) and I added a port to Torguards port forwarding. After starting the container again, the vpn seems to connect, then immediately fails to get the public ip, becomes unhealthy and gets stuck in a restart loop. So I removed the port again from the gluetun environment variables as well as from the torguard website, however it still doesn't work, and gets stuck in a reboot loop the same way as with the port forwarding. Any ideas what could be causing this?

**Comment by cplieger** [comment]
Had the same issue and switched away from gluetun to use hotio's qbittorrent version

**Comment by Gaurhoth** [comment]
I've had same problem. After using Gluetun to connect to Torguard for a couple of days, it mysteriously stops working. I've worked extensively with Torguard support and they have been clueless. Using tcpdump, I can confirm I never receive a response from torguard to attempts to handshake. 

After it stops working, even using the torguard client to attempt both wireguard and openvpn connections subsequently fail. Tried UDP and TCP (for openvpn) ports. No luck.

In fact, even using netcat to try and connect to torguard IPs on TCP (for ports that listen on tcp for openvpn), it's obvious that there's a firewall/IDS just straight up blocking my server's public IP. Change the public IP of the server and it works again for a couple of days before it stops working. 

Connections to other VPN providers works fine, but none of the Torguard endpoints work.

I suspect some behaviour of gluetun is triggering a blacklist by TorGuard - but no idea what it is.
